features/dapp_develop/dapp_develop_test.go

```diff
@@ -0,0 +1,229 @@
+       t.Fatal("Failed to pass all soroban dapp e2e tests")
+    }
+}
+
+func newTestConfig (e2eConfig *e2e.E2EConfig) *testConfig{
+    return &testConfig{
+        E2EConfig: e2eConfig,
+    }
+}
+
+func compileContract(ctx context.Context, contractExamplesSubPath string) error {
+    
+    testConfig := ctx.Value(e2e.TestConfigContextKey).(*testConfig)
+
+    contractWorkingDirectory := fmt.Sprintf("%s/soroban_examples", testConfig.TestWorkingDir)
+    envCmd := cmd.NewCmd("git" ,"clone", testConfig.E2EConfig.SorobanExamplesRepoURL, contractWorkingDirectory) 
+
+    status, _, err := e2e.RunCommand(envCmd, testConfig.E2EConfig)
+
+    if status != 0 || err != nil {
+        return fmt.Errorf("git clone of soroban example contracts from %s had error %v, %v", testConfig.E2EConfig.SorobanExamplesRepoURL, status, err)
+    }
+
+    envCmd = cmd.NewCmd("/bin/sh", "-c", fmt.Sprintf("cd %s; git checkout %s", contractWorkingDirectory, testConfig.E2EConfig.SorobanExamplesGitHash)) 
+
+    status, _, err = e2e.RunCommand(envCmd, testConfig.E2EConfig)
+
+    if status != 0 || err != nil {
+        return fmt.Errorf("git checkout %v of sample contracts repo %s had error %v, %v", testConfig.E2EConfig.SorobanExamplesGitHash, testConfig.E2EConfig.SorobanExamplesRepoURL, status, err)
+    }
+
+    envCmd = cmd.NewCmd("/bin/sh", "-c", fmt.Sprintf("cd %s/%s; cargo build --config net.git-fetch-with-cli=true --target wasm32-unknown-unknown --release", contractWorkingDirectory, contractExamplesSubPath)) 
+
+    status, _, err = e2e.RunCommand(envCmd, testConfig.E2EConfig)
+
+    if status != 0 || err != nil {
+        return fmt.Errorf("cargo build of sample contract %v/%v had error %v, %v", testConfig.E2EConfig.SorobanExamplesRepoURL, contractExamplesSubPath, status, err)
+    }
+
+    return nil
+}
+
+func deployContract(ctx context.Context, compiledContractFileName string) error {
+	
+    testConfig := ctx.Value(e2e.TestConfigContextKey).(*testConfig)
+    contractWorkingDirectory := fmt.Sprintf("%s/soroban_examples", testConfig.TestWorkingDir)
+
+    envCmd := cmd.NewCmd("/bin/sh", "-c", 
+        fmt.Sprintf("soroban deploy --wasm ./%s/target/wasm32-unknown-unknown/release/%s --rpc-url %s --secret-key %s --network-passphrase \"%s\"", 
+    	    contractWorkingDirectory, 
+    	    compiledContractFileName,
+            testConfig.E2EConfig.TargetNetworkRPCURL,
+            testConfig.E2EConfig.TargetNetworkSecretKey,
+            testConfig.E2EConfig.TargetNetworkPassPhrase)) 
+ 
+    status, stdOut, err := e2e.RunCommand(envCmd, testConfig.E2EConfig)
+
+    if status != 0 || err != nil {
+    	return fmt.Errorf("soroban cli deployment of example contract %s had error %v, %v", compiledContractFileName, status, err)
+    }
+
+    if len(stdOut) < 1 {
+    	return fmt.Errorf("soroban cli deployment of example contract %s returned no contract id", compiledContractFileName)
+    }
+
+    testConfig.DeployedContractId = stdOut[0]
+	return nil
```

Comment:
gofmt on this repo?

Reply:
done, thx - https://github.com/stellar/system-test/pull/1/commits/f8f43513c1f9386d34fd50c7786c182fe044b87f